sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1458,7 +1458,7 @@ def test_k_and_radius_neighbors_duplicates():
 
         # Mask the first duplicates when n_duplicates > n_neighbors.
         X = np.ones((3, 1))
-        nn = neighbors.NearestNeighbors(n_neighbors=1)
+        nn = neighbors.NearestNeighbors(n_neighbors=1, algorithm='brute')
```

Comment:
Why do you need this change?

Reply:
This part of the test used `algorithm='auto'`default, which selected `brute` with the original heuristic. The new heuristic selects `kd_tree` instead. As `brute` doesn't always output the same indices of duplicates than tree-based methods when `n_duplicates > n_neighbors`, the test would fail with the new heuristic.

I don't really understand why they don't all output the same indices of duplicates, but in order to preserve the original behavior of the test, I set `algorithm='brute'`